inference-engine/thirdparty/clDNN/src/graph_optimizer/prepare_padding.cpp

```diff
@@ -17,10 +17,30 @@ void prepare_padding::run(program& p) {
     if (output_size_handling_enabled) {
         // Prepare upper padding for primitives that support output_size parameter.
         for (const auto& node : p.get_processing_order()) {
+            if (node->get_dependencies().empty())
+                continue;
+
+            if (node->get_dependency(0).is_type<data>())
+                continue;
+
             // Padded offsets aren't supported by onednn kernels
             if (node->get_preferred_impl_type() == impl_types::onednn)
                 continue;
 
+            auto add_required_padding = [&p](program_node& node, padding& needed_padding) {
+                // Add extra reorder for cldnn primitive to handle required padding if needed
+                auto& input = node.get_dependency(0);
+                if (input.get_preferred_impl_type() == impl_types::onednn &&
+                    node.get_preferred_impl_type() == impl_types::ocl &&
+                    static_cast<bool>(needed_padding)) {
+                    auto new_reorder = std::make_shared<reorder>(node.id() + "_padding_reorder_for_" + input.id(), input.id(), input.get_output_layout());
+                    auto& new_reorder_node = p.get_or_create(new_reorder);
+                    p.add_intermediate(new_reorder_node, node, input);
+                }
+
+                p.apply_needed_padding(node, node.get_dependency(0), needed_padding);
```

Comment:
Could you add an assertion in cldnn conv itself to confirm that it has proper padding for input and output?

Reply:
it is always there. so mark it as resolved.